syntax = "proto3";

package locationservice;
option java_package = "com.air.lbs.protobuf";

message Car {
    string id = 1;
    repeated Location history = 2;
    Location last_position = 3;
}

message Location {
    double latitude = 1;
    double longitude = 2;
    double time = 3;
}

message PutRequest {
    string id = 1;
    double latitude = 2;
    double longitude = 3;
}

message GetRequest {
    string id = 1;
}

message GetResponse {
    double latitude = 1;
    double longitude = 2;
}

message SearchRequest {
    double latitude = 1;
    double longitude = 2;
    double radius = 3;
}

message SearchResponse {
    repeated Car cars = 1;
}

message HistoryRequest {
    string id = 1;
    string start_time = 2;
    string end_time = 3;
}

service LocationBaseService {
    rpc put(PutRequest) returns (Car) {};
    rpc get(GetRequest) returns (GetResponse) {};
    rpc search(SearchRequest) returns (SearchResponse) {};
    rpc history(HistoryRequest) returns (Car) {};
}